lib/experimental/fonts-library/class-wp-font-family-utils.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Fonts Family Utils class.
+ *
+ * This file contains utils fot Font Family class.
+ *
+ * @package    Gutenberg
+ * @subpackage Fonts Library
+ * @since      X.X.X
+ */
+
+if ( class_exists( 'WP_Font_Family_Utils' ) ) {
+	return;
+}
+
+/**
+ * A class of utilities for working with the Font Library.
+ */
+class WP_Font_Family_Utils {
+
+	/**
+	 * Generates a filename for a font face asset.
+	 *
+	 * Creates a filename for a font face asset using font family, style, weight and extension information.
+	 *
+	 * @param array  $font_face The font face array containing 'fontFamily', 'fontStyle', and 'fontWeight' attributes.
+	 * @param string $url The URL of the font face asset, used to derive the file extension.
+	 * @param int    $i Optional counter for appending to the filename, default is 1.
+	 * @return string The generated filename for the font face asset.
+	 */
+	public static function get_filename_from_font_face( $font_face, $url, $i = 1 ) {
+		$extension = pathinfo( $url, PATHINFO_EXTENSION );
+		$filename  = sanitize_title( "{$font_face['fontFamily']}_{$font_face['fontStyle']}_{$font_face['fontWeight']}" );
+		if ( $i > 1 ) {
+			$filename .= "_{$i}";
+		}
+		return "{$filename}.{$extension}";
+	}
+
+	/**
+	 * Merges two fonts and their font faces.
+	 *
+	 * @param array $font1 The first font to merge.
+	 * @param array $font2 The second font to merge.
+	 *
+	 * @return array|WP_Error The merged font or WP_Error if the fonts have different slugs.
+	 */
+	public static function merge_fonts_data( $font1, $font2 ) {
+		if ( $font1['slug'] !== $font2['slug'] ) {
+			return new WP_Error( 'fonts_must_have_same_slug', __( 'Fonts must have the same slug to be merged.', 'gutenberg' ) );
+		}
+		$font_faces_1      = isset( $font1['fontFace'] ) ? $font1['fontFace'] : array();
+		$font_faces_2      = isset( $font2['fontFace'] ) ? $font2['fontFace'] : array();
```

Comment:
@matiasbenedetto @hellofromtonya Is Gutenberg planning to bump its minimum version to PHP 7? If so, can we use the null coalescing operator here?

Reply:
>If so, can we use the null coalescing operator here?

Hey @costdev Gutenberg did bump to 7.0. But need to wait on adopting 7.0 changes. Waiting on the PHP 7.0 Code Standards Make/Core proposal.